### ISSUE TITLE
Inflector makes wrong snake case transformation

### ISSUE THREAD

**Issue by xepozz** [yiisoft/strings#92]
I expect that `$inflector->toSnakeCase($str)` will do the following transformations:
```
HTML => html
createMyDTO => create_my_dto
```
But it does `HTML` => `h_t_m_l`. See https://github.com/yiisoft/strings/blob/master/tests/InflectorTest.php#L170

### Additional info

| Q                | A
| ---------------- | ---
| Version          | 1.0.?
| PHP version      | 
| Operating system |

**Comment by samdark** [comment]
```
Aaz
aAz
A az

=>

aaz
a_az
a_az
```

**Comment by cebe** [comment]
Found the related discussion for yii2: https://github.com/yiisoft/yii2/pull/4166

**Comment by Tigrov** [comment]
There is `$inflector->pascalCaseToId($str)` which works as expected for the case.

https://github.com/yiisoft/strings/blob/master/tests/InflectorTest.php#L519

**Comment by vjik** [comment]
We can implement current issue by two stage.

1) Add to method `toSnakeCase` parameter `bool $strict` with default value `true` that pass to `pascalCaseToId`. This will be minor change and allow use `toSnakeCase` in two modes.

2) Before major release chane default value of `$strict` parameter to `false`.

**Comment by vjik** [comment]
Closed favor in #112
